File: jasmine/lib/jasmine/utils/hmac.ex
defmodule Jasmine.Utils.HMac do
  require Logger

  def sign(p) do
    {:ok, key} = Application.get_env(:jasmine, HMac)[:key] |> Base.decode64()
    :crypto.mac(:hmac, :sha512, key, p)
  end

  def verify?(c, p) do
    c == sign(p)
  end
end

File: petunia/src/hmac.zig
const std = @import("std");

const Hmac = struct {
    fn sign(_: *Hmac) void {
        // TODO
    }
    fn verify(_: *Hmac) void {
        // TODO
    }
};

test "hmac sha512" {
    try std.testing.expect(1 + 1 == 2);
}
